compaction/table_state.hh

```diff
@@ -47,6 +48,13 @@ public:
     virtual bool is_auto_compaction_disabled_by_user() const noexcept = 0;
     virtual const tombstone_gc_state& get_tombstone_gc_state() const noexcept = 0;
     virtual compaction_backlog_tracker& get_backlog_tracker() = 0;
+
+    const compaction::owned_ranges_ptr& get_owned_ranges_ptr() const noexcept {
+        return _owned_ranges_ptr;
+    }
+    void set_owned_ranges_ptr(compaction::owned_ranges_ptr p) noexcept {
+        _owned_ranges_ptr = std::move(p);
+    }
```

Comment:
Shouldn't this class remain a pure interface?

Reply:
I think we can move _owned_ranges_ptr into compaction_manager::compaction_state as it's per-table metadata that can be used for building the descriptor.

then retrieve_owned_ranges_if_required() becomes a compaction manager function, which conditionally sets descriptor's owned ranges field.

Reply:
fixed in next v2

Reply:
done in v2